data_processors/pipeline/lambdas/tumor_normal.py

```diff
@@ -0,0 +1,213 @@
+logger.setLevel(logging.INFO)
+
+
+def sqs_handler(event, context):
+    """
+    Unpack body from SQS wrapper.
+    SQS event payload dict:
+    {
+        'Records': [
+            {
+                'messageId': "11d6ee51-4cc7-4302-9e22-7cd8afdaadf5",
+                'body': "{\"JSON\": \"Formatted Message\"}",
+                'messageAttributes': {},
+                'md5OfBody': "e4e68fb7bd0e697a0ae8f1bb342846b3",
+                'eventSource': "aws:sqs",
+                'eventSourceARN': "arn:aws:sqs:us-east-2:123456789012:fifo.fifo",
+            },
+            ...
+        ]
+    }
+
+    Details event payload dict refer to https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html
+    Backing queue is FIFO queue and, guaranteed delivery-once, no duplication.
+
+    :param event:
+    :param context:
+    :return:
+    """
+    messages = event['Records']
+
+    results = []
+    for message in messages:
+        job = libjson.loads(message['body'])
+        results.append(handler(job, context))
+
+    return {
+        'results': results
+    }
+
+
+def handler(event, context) -> dict:
+    """event payload dict
+    {
+        "subject_id": "SUBJECT_ID",
+        "fastq_list_rows": [{
+            "rgid": "index1.index2.lane",
+            "rgsm": "sample_name",
+            "rglb": "UnknownLibrary",
+            "lane": int,
+            "read_1": {
+              "class": "File",
+              "location": "gds://path/to/read_1.fastq.gz"
+            },
+            "read_2": {
+              "class": "File",
+              "location": "gds://path/to/read_2.fastq.gz"
+            }
+        }],
+        "tumor_fastq_list_rows": [{
+            "rgid": "index1.index2.lane",
+            "rgsm": "sample_name",
+            "rglb": "UnknownLibrary",
+            "lane": int,
+            "read_1": {
+              "class": "File",
+              "location": "gds://path/to/read_1.fastq.gz"
+            },
+            "read_2": {
+              "class": "File",
+              "location": "gds://path/to/read_2.fastq.gz"
+            }
+        }],
+        "output_file_prefix": "SAMPLEID_LIBRARYID",
+        "output_directory": "SAMPLEID_LIBRARYID"
+    }
+
+    :param event:
+    :param context:
+    :return: workflow db record id, wfr_id, sample_name in JSON string
+    """
+
+    logger.info(f"Start processing {WorkflowType.TUMOR_NORMAL.name} event")
+    logger.info(libjson.dumps(event))
+
+    # Extract name of sample and the fastq list rows
+    subject_id = event['subject_id']
+    output_file_prefix = event['output_file_prefix']
+    output_directory = event['output_directory']
+    fastq_list_rows = event['fastq_list_rows']
+    tumor_fastq_list_rows = event['tumor_fastq_list_rows']
+
+    # Set workflow helper
+    wfl_helper = WorkflowHelper(WorkflowType.TUMOR_NORMAL)
+
+    # Read input template from parameter store and populate values
+    input_template = libssm.get_ssm_param(wfl_helper.get_ssm_key_input())
```

Comment:
Do we have this input template ready to go? Can we tell if the event attributes `fastq_list_rows` and `tumor_fastq_list_rows` are json-strings or dicts/objects?

Reply:
The input template is stored in [SSM parameter store](https://ap-southeast-2.console.aws.amazon.com/systems-manager/parameters/?region=ap-southeast-2&tab=Table#list_parameter_filters=Name:Contains:workflow%2Ftumor) analogous to the other workflows.

The fastq_list_rows are JSON serialisable dicts (generated from the FastqListRow objects).

Reply:
Yep, just thought I'd clarify given we have the germline under [main.tf](https://github.com/umccr/infrastructure/blob/5e08f2028dee2754da12ba92c92516b6dc06380c/terraform/stacks/umccr_data_portal/workflow/main.tf#L99-L121). I can add the somatic to this list too? Or are you happy for it just to be registered manually in ssm for now?

Reply:
Ah, yes! Good point!
Please add there. I just put it in SSM manually and forgot about it.

Reply:
Added SSM parameters to TF in https://github.com/umccr/infrastructure/commit/4d77bc2585ec1bd2358107e05e6c431d581ee19f